app/scripts/migrations/144.ts

```diff
@@ -0,0 +1,300 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, Hex, isObject, RuntimeObject } from '@metamask/utils';
+import { nanoid } from 'nanoid';
+import type {
+  PermissionConstraint,
+  PermissionControllerSubjects,
+} from '@metamask/permission-controller';
+import { SnapEndowments } from '@metamask/snaps-rpc-methods';
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+  addPermittedEthChainId,
+  Caip25CaveatValue,
+} from '@metamask/multichain';
+
+type GenericPermissionControllerSubject =
+  PermissionControllerSubjects<PermissionConstraint>[string];
+
+export const version = 144;
+
+/**
+ * A map of the networks built into the extension at the time of this migration
+ * to their chain IDs.
+ *
+ * @see https://github.com/MetaMask/metamask-extension/blob/5b5c04a16fb7937a6e9d59b1debe4713978ef39d/shared/constants/network.ts#L535
+ */
+const BUILT_IN_NETWORKS: ReadonlyMap<string, Hex> = new Map([
+  ['sepolia', '0xaa36a7'],
+  ['mainnet', '0x1'],
+  ['linea-sepolia', '0xe705'],
+  ['linea-mainnet', '0xe708'],
+]);
+
+/**
+ * This migration adds the `endowment:caip25` permission to all Snaps
+ * that have the `endowment:ethereum-provider` permission, and sets the selected
+ * chain ID for each Snap to the current selected network chain ID.
+ *
+ * This is necessary following the Amon Hen v2 changes to the network controller
+ * and other related controllers, as the `endowment:permitted-chains` permission
```

Comment:
```suggestion
 * and other related controllers, as the `endowment:caip25` permission
```

These comments may need more updating though, since the CAIP25 permission is already added, we just wanna add the chain ID